onnx/numpy_helper.py

```diff
@@ -259,18 +255,11 @@ def unpack_float4e2m1(
     Returns:
         A numpy array of float32 reshaped to dims.
     """
-    single_func = lambda x: subbyte.unpack_single_4bitx2(x, False)  # noqa: E731
-    func = np.frompyfunc(single_func, 1, 2)
-
-    res_high, res_low = func(data.ravel())
+    res_high, res_low = subbyte.unpack_single_4bitx2(data.ravel(), False)
```

Comment:
I would consider referring to https://github.com/microsoft/onnxscript/blob/87aee66c1c7eb1e53e4bb2f6bb32012978019d9c/onnxscript/ir/_type_casting.py#L31-L53 for unpacking. `unpack_single_4bitx2` seems unnecessary and potentially inefficient. 

Potentally:

1. Call unpack_uint4_as_uint8() to get a view of the float4 array
2. Call evaluate_float4e2m1_from_bits to cast to float32?

Feel free to take unpack_uint4_as_uint8 and move it subbyte.py as you see fit.

Reply:
I think `unpack_single_4bitx2` is doing the same thing as `unpack_uint4_as_uint8` in onnxscript?
It automatically broadcasts to the entire input array without need for `frompyfunc`.

Reply:
unpack_uint4_as_uint8 does a little bit more than unpack_single_4bitx2 in that it also handles reshaping etc. With that you can simplify the implementation of unpack_float4e2m1